Library/Formula/pcap_dnsproxy.rb

```diff
@@ -0,0 +1,51 @@
+class PcapDnsproxy < Formula
+  desc "A powerful DNS proxy designed to anti DNS spoofing"
+  homepage "https://github.com/chengr28/Pcap_DNSProxy"
+  url "https://github.com/chengr28/Pcap_DNSProxy/archive/v0.4.2.tar.gz"
+  sha256 "0737b16283e84bc94f63066f448ba84416141b8d5a86b5e01f1f97164d9d44f2"
+  head "https://github.com/chengr28/Pcap_DNSProxy.git"
+
+  depends_on :xcode => :build
+  depends_on "libsodium" => :build
+
+  def install
+    ln_s "#{Formula["libsodium"].opt_lib}/libsodium.a", "#{buildpath}/Source/LibSodium/LibSodium_Mac.a"
+    xcodebuild "-project", "./Source/KeyPairGenerator.xcodeproj", "-target", "KeyPairGenerator", "-configuration", "Release", "SYMROOT=build"
+    xcodebuild "-project", "./Source/Pcap_DNSProxy.xcodeproj", "-target", "Pcap_DNSProxy", "-configuration", "Release", "SYMROOT=build"
+
+    bin.install "Source/build/Release/KeyPairGenerator"
+    bin.install "Source/build/Release/Pcap_DNSProxy"
+    (etc/"pcap_DNSproxy").install Dir["Source/ExampleConfig/*.{ini,txt}"]
+  end
+
+  test do
+    system bin/"Pcap_DNSProxy", "--version"
+  end
+
+  plist_options :startup => true, :manual => "sudo #{HOMEBREW_PREFIX}/opt/pcap_dnsproxy/bin/Pcap_DNSProxy -c #{HOMEBREW_PREFIX}/etc/pcap_dnsproxy/"
```

Comment:
This software listen a localhost port and work as a DNS server, it proxy the DNS resolve request and filter the polluted DNS response to anti DNS spooning. Then set the DNS server to 127.0.0.1, we can avoid DNS spooning.

Sure we can use it without `sudo`, but if you want to listen port < 1024, you need a root privilege.
Unfortunately you can't use port other than 53 as OS X's DNS server in system preference.

This is why the `sudo` is necessary.